tests/models/test_NaiveBayes.py

```diff
@@ -0,0 +1,141 @@
+import shutil
+import tempfile
+import unittest
+
+import numpy as np
+import scipy.stats
+from pymc3 import summary
+from sklearn.model_selection import train_test_split
+
+from pymc3_models.exc import  PyMC3ModelsError
+from pymc3_models import GaussianNaiveBayes
+
+
+
+class GaussianNaiveBayesTestCase(unittest.TestCase):
+    def setUp(self):
+        """ Set up a test case with synthetic data.
+        """
+
+        self.num_cats = 3
+        self.num_features = 10
+        self.num_samples = 100000
+        
+        # Generate data
+        ## Priors
+        alpha = np.ones(self.num_cats)
+        pi = np.random.dirichlet(alpha)
+        mu = np.random.normal(0, 100, size=(self.num_cats, self.num_features))
+        sigma = scipy.stats.halfnorm(loc=0, scale=100).rvs(size=(self.num_cats, self.num_features))
+        ## Data
+        Y = np.random.choice(range(self.num_cats), self.num_samples, p=pi)
+        x_vectors = []
+        for i in Y:
+            x_vectors.append(np.random.normal(mu[i], sigma[i]))
+        X = np.vstack(x_vectors)
+        
+        
+        # Split into train/test sets
+        split = train_test_split(X, Y, test_size=0.4)
+        self.X_train = split[0] 
+        self.X_test= split[1]
+        self.Y_train = split[2]
+        self.Y_test = split[3]
+
+        self.test_GNB = GaussianNaiveBayes()
+        self.test_dir = tempfile.mkdtemp()
+        
+        # Fit the model once and for all
+        self.test_GNB.fit(self.X_train, self.Y_train, minibatch_size=2000)
+
+    def tearDown(self):
+        """ Tear down the testing environment.
+        """
+        shutil.rmtree(self.test_dir)
+
+
+# Test fit
+class GaussianNaiveBayesFitTestCase(GaussianNaiveBayesTestCase):
+    def test_fit_returns_correct_model(self):
+        """ Test the model initialization and fit
+
+        Currently, only the sign of infered parameters is checked
+        against the sign of the parameters used to generate the data.
+
+        This is, however, not ideal. It is necessary to find better 
+        strategies to test probabilistic code.
+        """
+
+        print('') # Berk
+        
+        # Check that the model correctly infers dimensions
+        self.assertEqual(self.num_cats, self.test_GNB.num_classes)
+        self.assertEqual(self.num_samples, self.test_GNB.num_samples)
+        self.assertEqual(self.num_features, self.test_GNB.num_features)
+
+        # TODO: How do you write tests for a stochastic mode?
+        # TODO: Diagnose the sampling with a reasonable sampling size?
+        np.testing.assert_equal(
+            np.sign(self.pi),
+            np.sign(self.test_GNB.trace['pi'].mean(axis=0))
+            )
+
+        np.testing.assert_equal(
+            np.sign(self.sigma),
+            np.sign(self.test_GNB.trace['sigma'].mean(axis=0))
+            )
+
+
+class GaussianNaiveBayesPredictProbaTest(GaussianNaiveBayesTestCase):
+    """ Test the predict_proba() method.
```

Comment:
Since your testcases are very explicitly named (yea!), you don't need docstrings.